src/libraries/oracle/OracleConsulter.sol

```diff
@@ -4,29 +4,46 @@ pragma solidity ^0.8.0;
 import "./OracleLibrary.sol";
 
 library OracleConsulter {
-    struct PriceConsultancyParams {
-        address pool;
-        uint32 period;
-        uint128 baseAmount;
-        address baseToken;
-        address quoteToken;
-    }
-
-    function consultPriceAtTick(PriceConsultancyParams calldata _params)
-        external
-        view
-        returns (uint256 twap)
-    {
-        (int24 timeWeightedAverageTick, ) = OracleLibrary.consult(
-            _params.pool,
-            _params.period
-        );
+    function consult(
+        address pool,
+        uint32 period,
+        uint128 baseAmount,
+        address baseToken,
+        address quoteToken
+    ) external view returns (uint256 twap) {
+        (int24 timeWeightedAverageTick, ) = OracleLibrary.consult(pool, period);
 
         twap = OracleLibrary.getQuoteAtTick(
             timeWeightedAverageTick,
-            _params.baseAmount,
-            _params.baseToken,
-            _params.quoteToken
+            baseAmount,
+            baseToken,
+            quoteToken
+        );
+    }
+
+    function consult(
+        address[] calldata pools,
+        address[] calldata tokens,
+        uint32 period,
+        uint128 baseAmount
+    ) external view returns (uint256) {
+        uint256 poolsLength = pools.length;
+        int24[] memory tickRoute = new int24[](poolsLength);
+
+        for (uint256 i = 0; i < poolsLength; i++) {
+            (tickRoute[i], ) = OracleLibrary.consult(pools[i], period);
+        }
+        int256 chainedMeanTick = OracleLibrary.getChainedPrice(
+            tokens,
+            tickRoute
         );
+
+        return
+            OracleLibrary.getQuoteAtTick(
+                int24(chainedMeanTick),
```

Comment:
What are your conclusions on whether casting to int24 is not losing information?

Reply:
I opened a [ticket](https://github.com/Uniswap/v3-periphery/issues/325) on Uni v3 periphery's repo. Haven't gotten a response, but I updated the chained consulter test in my oracle tests for a wider range of inputs.

I also added a `require` to the `OracleConsulter` lib to check for overflows just in case.

If I receive a response on my ticket or independently prove that overflow is not possible or that it does not affect the validity of the consulter, then I will remove the check to save on gas.